metaflow/metadata_provider/metadata.py

```diff
@@ -672,6 +673,39 @@ def _register_system_metadata(self, run_id, step_name, task_id, attempt):
         if metadata:
             self.register_metadata(run_id, step_name, task_id, metadata)
 
+    @classmethod
+    def filter_tasks_by_metadata(
+        cls,
+        flow_id: str,
+        run_id: str,
+        query_step: str,
+        field_name: str,
+        field_value: str,
+    ) -> List[str]:
+        """
+        Filter tasks by metadata field and value, and returns the list of task_ids
+        that satisfy the query.
+
+        Parameters
+        ----------
+        flow_id : str
+            Flow id, that the run belongs to.
+        run_id: str
+            Run id, together with flow_id, that identifies the specific Run whose tasks to query
+        query_step: str
+            Step name to query tasks from
+        field_name: str
+            Metadata field name to query
+        field_value: str
+            Metadata field value to query
+
+        Returns
+        -------
+        List[str]
```

Comment:
Is this always task_ids? Since we are making this visible to the user, it should be task_name if the user visible task identifier is the name. Basically if the task is known as 123 to the user, it should return that but if it is known as task-123, it should return that.